pkgs/build-support/kernel/compress-firmware-xz.nix

```diff
@@ -7,14 +7,18 @@ let
 in
 
 runCommand "${firmware.name}-xz" args ''
+  edids=(-path 'lib/firmware/edid/*')
   mkdir -p $out/lib
   (cd ${firmware} && find lib/firmware -type d -print0) |
-      (cd $out && xargs -0 mkdir -v --)
-  (cd ${firmware} && find lib/firmware -type f -print0) |
+      (cd $out && xargs -0 mkdir -pv --)
+  (cd ${firmware} && find lib/firmware -type f -not "''${edids[@]}" -print0) |
```

Comment:
Why does this need to be a bash array thing?

Reply:
It's just my preference to use array whenever there is more than 1 argument and running inside bash.

There is no chance to misuse array as opposed to expanding multi-argument variable.

Reply:
But you leave the globbing up to `find`, no? So there's no spaces in paths.

I think it'd make more sense for readability to just "constant-propagate" the -path 'lib/firmware/edid/*' into the three occurences, and add a comment in front, stating edids are excluded from compression.

Reply:
> But you leave the globbing up to `find`, no? So there's no spaces in paths.
This is not really globbing, just a prefix, `find` was not happy about `-path lib/firmware/edid`

Reply:
But it'd be happy about ' -path 'lib/firmware/edid/*', no?

> I think it'd make more sense for readability to just "constant-propagate" the -path 'lib/firmware/edid/*' into the three occurences, and add a comment in front, stating edids are excluded from compression.

I think that's approach is simpler than these bash arrays. We can revisit once there's more than a single element, but let's keep this simple for now, happy to merge then.

Reply:
well, let it be, I left 2 notes on top of the file.